src/DIRAC/WorkloadManagementSystem/Client/JobState/JobState.py

```diff
@@ -37,6 +39,7 @@ def checkDBAccess(cls):
     def __init__(self, jid, source="Unknown"):
         self.__jid = jid
         self.__source = str(source)
+        self.log = JobLogger(gLogger, jid)
```

Comment:
It is unused?

Reply:
I fixed it by using a subLogger for each optimizer and for the optimizerAdministrator